src/main/java/com/google/cloud/tools/maven/run/Runner.java

```diff
@@ -82,23 +80,16 @@ public void runAsync(int startSuccessTimeout) throws MojoExecutionException {
     runMojo.getLog().info("Use the 'mvn appengine:stop' command to stop the server.");
   }
 
-  private Path getAppDir() {
-    Build build = runMojo.getMavenProject().getBuild();
-    return Paths.get(build.getDirectory()).resolve(build.getFinalName());
-  }
-
   static final String NON_STANDARD_APPLICATION_ERROR =
       "\nCould not find appengine-web.xml all services, perhaps you need to run "
           + "'mvn package appengine:run/start'."
           + "\nDev App Server only supports appengine-web.xml based Java applications.";
 
   @VisibleForTesting
   List<Path> processServices() throws MojoExecutionException {
-    Path appDir = getAppDir();
     List<Path> services = runMojo.getServices();
-    if (services == null || services.isEmpty()) {
-      return Collections.singletonList(appDir);
-    }
+
+    Preconditions.checkState(services != null && !services.isEmpty(), "No services found to run");
```

Comment:
I prefer two check conditions with two different error messages, so that in case of a failure at runtime, we can easily figure out which condition was violated.

Reply:
Technically this is unreachable. `getServices()` cannot (and should not) return `null or empty`. This more of an assertion than anything.

Reply:
changed this anyway.